src/open_inwoner/openzaak/statuses.py

```diff
@@ -33,7 +34,44 @@ def fetch_status_history(case_url: str) -> List[Status]:
     return statuses
 
 
-def fetch_status_types(zaaktype: str) -> List[StatusType]:
+def fetch_specific_statuses(status_urls: List[str]) -> List[Status]:
+    client = build_client("zaak")
+
+    if client is None:
+        return []
+
+    with parallel() as executor:
+        _statuses = executor.map(
+            lambda url: client.retrieve("status", url=url),
+            status_urls,
+        )
+
+    statuses = factory(Status, list(_statuses))
+
+    return statuses
+
+
+def fetch_status_types() -> List[StatusType]:
+    client = build_client("catalogi")
+
+    if client is None:
+        return []
+
+    try:
+        response = get_paginated_results(client, "statustype")
+    except RequestException as e:
+        logger.exception("exception while making request", exc_info=e)
+        return []
+    except ClientError as e:
+        logger.exception("exception while making request", exc_info=e)
+        return []
+
+    status_types = factory(StatusType, response)
+
+    return status_types
+
+
+def fetch_specific_status_types(case_type: str) -> List[StatusType]:
```

Comment:
It looks like `fetch_specific_status_types` and `fetch_status_types` are basically the same function, the only difference is the input parameter. I think these functions can be merged into one  with `case_type` as an optional parameter. 
It will help to make code DRYer